File: src/main/java/kitchenpos/deliveryorders/infra/MenuClientImpl.java
package kitchenpos.deliveryorders.infra;

import kitchenpos.deliveryorders.domain.MenuClient;
import kitchenpos.menus.tobe.domain.menu.Menu;
import kitchenpos.menus.tobe.domain.menu.MenuRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Service
public class MenuClientImpl implements MenuClient {
    private final MenuRepository menuRepository;

    public MenuClientImpl(MenuRepository menuRepository) {
        this.menuRepository = menuRepository;
    }

    @Override
    public Optional<Menu> findById(UUID id) {
        return menuRepository.findById(id);
    }

    @Override
    public int countAllByIdIn(List<UUID> ids) {
        return menuRepository.findAllByIdIn(ids).size();
    }
}

File: src/main/kotlin/kitchenpos/tobe/eatinorder/infra/MenuClientImpl.kt
package kitchenpos.tobe.eatinorder.infra

import kitchenpos.tobe.eatinorder.domain.MenuClient
import kitchenpos.tobe.eatinorder.domain.dto.MenuData
import kitchenpos.tobe.menu.domain.repository.MenuRepositoryV2
import org.springframework.stereotype.Service
import java.util.*

@Service
class MenuClientImpl(
    private val menuRepository: MenuRepositoryV2,
) : MenuClient {
    override fun getMenus(menuIds: List<UUID>): List<MenuData> {
        val menus = menuRepository.findAllByIdIn(menuIds)

        if (menus.size != menuIds.size) {
            throw IllegalArgumentException("메뉴가 존재하지 않습니다.")
        }

        return menus.map {
            MenuData(
                id = it.id,
                displayed = it.displayed,
                price = it.getPrice(),
            )
        }
    }
}
